node/actors/control_api.go

```diff
@@ -181,62 +180,198 @@ func (api *ControlAPI) subscribe() error {
+	if err != nil {
+		api.logger.Error("Failed to locate agent actor", slog.String("type", req.WorkloadType), slog.Any("error", err))
+		respondEnvelope(m, RunResponseType, 500, nil, fmt.Sprintf("failed to locate [%s] agent actor: %s", req.WorkloadType, err))
+		return
+	}
+
+	askResp, err := api.self.Ask(ctx, agent, startRequestToProto(req))
+	if err != nil {
+		api.logger.Error("Failed to start workload", slog.Any("error", err))
+		respondEnvelope(m, InfoResponseType, 500, nil, fmt.Sprintf("Failed to start workload: %s", err))
+		return
+	}
+
+	protoResp, ok := askResp.(*actorproto.WorkloadStarted)
+	if !ok {
+		api.logger.Error("Start workload response from agent was not the correct type")
+		respondEnvelope(m, RunResponseType, 500, nil, "Agent returned the wrong data type")
+		return
+	}
+
+	respondEnvelope(m, RunResponseType, 200, startResponseFromProto(protoResp), "")
 }
 
-func (api *ControlAPI) handleInfo(m *nats.Msg) {
+func (api *ControlAPI) handleUndeploy(m *nats.Msg) {
+	req := new(nodecontrol.StopWorkloadRequestJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal undeploy request", slog.Any("error", err))
+		respondEnvelope(m, StopResponseType, 500, nil, fmt.Sprintf("failed to unmarshal undeploy request: %s", err))
+		return
+	}
 
-	req := &actorproto.QueryWorkloads{}
+	ctx := context.Background()
+	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
+	if err != nil {
+		api.logger.Error("Failed to locate agent supervisor actor", slog.Any("error", err))
+		respondEnvelope(m, StopResponseType, 500, nil, fmt.Sprintf("failed to locate agent supervisor actor: %s", err))
+		return
+	}
+
+	askResp, err := api.self.Ask(ctx, agentSuper, stopRequestToProto(req))
+	if err != nil {
+		api.logger.Error("Failed to locate agent supervisor actor", slog.Any("error", err))
+		respondEnvelope(m, StopResponseType, 500, nil, fmt.Sprintf("failed to locate agent supervisor actor: %s", err))
+		return
+	}
+
+	protoResp, ok := askResp.(*actorproto.WorkloadStopped)
+	if !ok {
+		api.logger.Error("Workload listing response from agent supervisor was not the correct type")
+		respondEnvelope(m, StopResponseType, 500, nil, "Agent supervisor returned the wrong data type")
+		return
+	}
+
+	respondEnvelope(m, StopResponseType, 200, stopResponseFromProto(protoResp), "")
+}
+
+func (api *ControlAPI) handleInfo(m *nats.Msg) {
 	ctx := context.Background()
 	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
 	if err != nil {
 		api.logger.Error("Failed to locate agent supervisor actor", slog.Any("error", err))
 		respondEnvelope(m, InfoResponseType, 500, nil, fmt.Sprintf("failed to locate agent supervisor actor: %s", err))
 		return
 	}
+
 	// Ask the agent supervisor for a list of all the workloads from all of its children
-	response, err := api.self.Ask(ctx, agentSuper, req)
+	response, err := api.self.Ask(ctx, agentSuper, new(actorproto.GetNodeInfo))
```

Comment:
agent super shouldn't respond to get node info... we should return a node info from inside this control api, and only ask the agent super for its list of workloads.

Information like the node Id, target xkey, etc, needs to come from node configuration on the api struct and not the agent super.